### PR TITLE
Add NextJS test app to CI

### PR DESCRIPTION
**Pull request by reidbarber**
Adds a NextJS test app to the `examples` directory and builds it with verdaccio.

## ✅ Pull Request Checklist:
- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Check the verdaccio comment on the latest commit for the link to the built NextJS app, along with the build stats.

### CODE REVIEW

File: .circleci/comment.js
Comment (reidbarber) [inline .circleci/comment.js:13]: TODO: revert before merge

File: .circleci/comment.js
Comment (reidbarber) [inline .circleci/comment.js:43]: TODO: revert before merge

File: .circleci/config.yml
Comment (reidbarber) [inline .circleci/config.yml:422]: TODO: revert before merge

File: .circleci/comment.js
Comment (reidbarber) [inline .circleci/comment.js:13]: TODO: revert before merge

File: .circleci/comment.js
Comment (ktabors) [inline .circleci/comment.js:53]: I wanted to look at these files, but I don't see a commit message. I thought the above allowed that.

File: examples/rsp-next-ts/pages/index.tsx
Comment (ktabors) [inline examples/rsp-next-ts/pages/index.tsx:123]: Does this project follow a different linter since these on on new lines?

File: examples/rsp-next-ts/pages/index.tsx
Comment (ktabors) [inline examples/rsp-next-ts/pages/index.tsx:345]: indeterminate?

File: examples/rsp-next-ts/pages/index.tsx
Comment (ktabors) [inline examples/rsp-next-ts/pages/index.tsx:72]: Would these be in a followup PR? LabeledValue, Color components, ActionBar, and DialogContainer? I'm guessing DnD and Table column resizing wouldn't be part of this coverage.

File: examples/rsp-next-ts/pages/index.tsx
Comment (ktabors) [inline examples/rsp-next-ts/pages/index.tsx:94]: With Next.js are we just confirming that they load or should we spread around some props (quiet, emphasized, etc.) throughout the file?

File: examples/rsp-next-ts/pages/index.tsx
Comment (ktabors) [inline examples/rsp-next-ts/pages/index.tsx:249]: Nested dialogs?

File: examples/rsp-next-ts/styles/Home.module.css
Comment (ktabors) [inline examples/rsp-next-ts/styles/Home.module.css:55]: Why do we have these styles?

File: examples/rsp-next-ts/styles/globals.css
Comment (ktabors) [inline examples/rsp-next-ts/styles/globals.css:5]: Do we need to do any font-family loading of the adobe fonts?

File: examples/rsp-next-ts/styles/Home.module.css
Comment (reidbarber) [reply examples/rsp-next-ts/styles/Home.module.css:55]: Just from create-next-app, I can remove the unused ones.

File: examples/rsp-next-ts/styles/globals.css
Comment (reidbarber) [reply examples/rsp-next-ts/styles/globals.css:5]: I think some people might use this as a template for their app, so probably not.

File: examples/rsp-next-ts/pages/index.tsx
Comment (reidbarber) [reply examples/rsp-next-ts/pages/index.tsx:123]: It has it's own eslint config (the default provided by `create-next-app`) and it was formatted with prettier, which I think is fine for this example IMO.

**Comment by snowystinger** [comment]
Found a weird styling issue, can you verify that it was pre-existing or a NextJS related issue? we can create a separate ticket for it, I doubt it's something you introduced.
Screen Shot 2022-10-06 at 10 36 35 AM

File: packages/dev/parcel-transformer-mdx-docs/MDXTransformer.js
Comment (reidbarber) [inline packages/dev/parcel-transformer-mdx-docs/MDXTransformer.js:330]: Some recent changes from main cause the docs-verdaccio build step to fail, so this fixes that.

**Comment by reidbarber** [comment]
> Found a weird styling issue, can you verify that it was pre-existing or a NextJS related issue? we can create a separate ticket for it, I doubt it's something you introduced. Screen Shot 2022-10-06 at 10 36 35 AM The track of the second slider is slightly higher than the first one

@snowystinger Just tried this locally in storybook and saw the same thing, so looks like this is an existing bug not related to NextJS.

File: examples/rsp-next-ts/pages/index.tsx
Comment (reidbarber) [reply examples/rsp-next-ts/pages/index.tsx:72]: I've added Color components. I think it's fine to do beta packages, but I won't do alpha packages since the APIs might change and this would require updating.

File: examples/rsp-next-ts/pages/index.tsx
Comment (reidbarber) [reply examples/rsp-next-ts/pages/index.tsx:101]: IMO any failures would likely happen on initial load or interactivity, so I think it's fine not to add every style variant.

**Review by snowystinger — DISMISSED** [review]
Checked out the app, looks good to me
I'd love more info on what happened in the MDXTransformer and why this is the right fix

otherwise, only one note, it seems to comment twice? it's not a big deal, this is a great starting point

I'll re-approve after reverts you noted happen, just wanted to express that I'm happy with the output nextjs app

**Review by ktabors — COMMENTED** [review]
Looking at the most recent "next-build-stats.txt" and I saw these characters (`â”Œ â—‹ /`) in the "route" section, which seems not ideal.

File: examples/rsp-next-ts/next.config.js
Comment (ktabors) [inline examples/rsp-next-ts/next.config.js:50]: why were the trailing commas on the last items added?

File: examples/rsp-next-ts/pages/index.tsx
Comment (ktabors) [inline examples/rsp-next-ts/pages/index.tsx:190]: In the color section we have ColorField rendering inline while the others are on newlines, seems by design.
image

**Review by ktabors — APPROVED** [review]
Looks good, thanks for the changes.

File: examples/rsp-next-ts/next.config.js
Comment (reidbarber) [reply examples/rsp-next-ts/next.config.js:50]: from prettier, just ran on all files in the project for consistency

**Comment by reidbarber** [comment]
> I'd love more info on what happened in the MDXTransformer and why this is the right fix

@snowystinger I think it has to do with the environment the verdaccio script runs on being different than the normal docs build. I couldn't find any specifics, but maybe we can fix that in the future.

**Comment by reidbarber** [comment]
> Looking at the most recent "next-build-stats.txt" and I saw these characters (â”Œ â—‹ /) in the "route" section, which seems not ideal.

@ktabors I think these characters are used for formatting and just get encoded strangely when outputted to the text file. 

Screen Shot 2022-10-07 at 12 56 33 PM

**Review by snowystinger — APPROVED** [review]

**Review by ktabors — APPROVED** [review]

**Pull request merged**
